Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/TapToPlace/TapToPlace.cs

```diff
@@ -0,0 +1,467 @@
+        private float surfaceNormalOffset = 0.0f;
+
+        /// <summary>
+        /// SurfaceNormalOffset is the offset between the game object's center and The default value for SurfaceNormalOffset is half the depth of a game object. 
+        /// </summary>
+        public float SurfaceNormalOffset
+        {
+            get => surfaceNormalOffset;
+            set
+            {
+                if (surfaceNormalOffset != value)
+                {
+                    surfaceNormalOffset = value;
+                }
+            }
+        }
+
+        [SerializeField]
+        [Tooltip("If true, the GameObjectToPlace will remain upright and in line with Vector3.up")]
+        private bool keepOrientationVertical = false;
+
+        /// <summary>
+        /// If true, the GameObjectToPlace will remain upright and in line with Vector3.up
+        /// </summary>
+        public bool KeepOrientationVertical
+        {
+            get => keepOrientationVertical;
+            set
+            {
+                if (keepOrientationVertical != value)
+                {
+                    keepOrientationVertical = value;
+                }
+            }
+        }
+
+
+        [SerializeField]
+        [Tooltip("If true, the spatial mesh will be visible while the object is in the placing state.")]
+        private bool spatialMeshVisible = true;
+
+        /// <summary>
+        /// If true, the spatial mesh will be visible while the object is in the placing state.
+        /// </summary>
+        public bool SpatialMeshVisible
+        {
+            get => spatialMeshVisible;
+            set
+            {
+                if (spatialMeshVisible != value)
+                {
+                    spatialMeshVisible = value; 
+                }
+            }
+        }
+
+
+
+        [SerializeField]
+        [Tooltip("")]
+        private SpatialAwarenessMeshDisplayOptions initialSpatialMeshDisplayOption;
+
+        /// <summary>
+        /// Get the spatial mesh visibility on start.  If the the spatial mesh is visible by default then do not toggle the
+        /// spatial mesh visibility while placing the game object.  Spatial mesh visibility refers to if the wire frame material is
+        /// on the spatial mesh or not.
+        /// </summary>
+        public SpatialAwarenessMeshDisplayOptions InitialSpatialMeshDisplayOption
+        {
+            get
+            {
+
+                IReadOnlyList<IMixedRealitySpatialAwarenessMeshObserver> observers = spatialAwarenessSystemAcecess.GetDataProviders<IMixedRealitySpatialAwarenessMeshObserver>();
+
+                foreach (IMixedRealitySpatialAwarenessMeshObserver observer in observers)
+                {
+                    return observer.DisplayOption;
+                }
+
+                // if the spatial awareness is not enabled
+                return SpatialAwarenessMeshDisplayOptions.None;   
```

Comment:
I believe this needs to be an array stored. You are only returning the first valid response option but later on in `ToggleSpatialMeshVisibility` you are apply that initial option from one observer to all possible observers

Reply:
SA dependency was removed